File: python-wrapper/src/python/model/base/LinkQualification.py
from typing import Union, Dict
from enum import Enum

class LinkQualification(Enum):
	Exact = "exact"
	Narrower = "narrower"
	Broader = "broader"
	Approximate = "approximate"
	Sequence = "sequence"
	Parent = "parent"
	Child = "child"
	RelatedCode = "relatedCode"
	LinkedPackage = "linkedPackage"
	RelatedService = "relatedService"
	InResponseTo = "inResponseTo"
	Replaces = "replaces"
	Transforms = "transforms"
	TransformsAndReplaces = "transformsAndReplaces"
	AppendsTo = "appendsTo"
	BasedOn = "basedOn"
	DerivedFrom = "derivedFrom"
	Device = "device"
	Focus = "focus"
	HasMember = "hasMember"
	Performer = "performer"
	Specimen = "specimen"
	ResultInterpreter = "resultInterpreter"
	Request = "request"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'LinkQualification':
		if data == "exact":
			return LinkQualification.Exact
		elif data == "narrower":
			return LinkQualification.Narrower
		elif data == "broader":
			return LinkQualification.Broader
		elif data == "approximate":
			return LinkQualification.Approximate
		elif data == "sequence":
			return LinkQualification.Sequence
		elif data == "parent":
			return LinkQualification.Parent
		elif data == "child":
			return LinkQualification.Child
		elif data == "relatedCode":
			return LinkQualification.RelatedCode
		elif data == "linkedPackage":
			return LinkQualification.LinkedPackage
		elif data == "relatedService":
			return LinkQualification.RelatedService
		elif data == "inResponseTo":
			return LinkQualification.InResponseTo
		elif data == "replaces":
			return LinkQualification.Replaces
		elif data == "transforms":
			return LinkQualification.Transforms
		elif data == "transformsAndReplaces":
			return LinkQualification.TransformsAndReplaces
		elif data == "appendsTo":
			return LinkQualification.AppendsTo
		elif data == "basedOn":
			return LinkQualification.BasedOn
		elif data == "derivedFrom":
			return LinkQualification.DerivedFrom
		elif data == "device":
			return LinkQualification.Device
		elif data == "focus":
			return LinkQualification.Focus
		elif data == "hasMember":
			return LinkQualification.HasMember
		elif data == "performer":
			return LinkQualification.Performer
		elif data == "specimen":
			return LinkQualification.Specimen
		elif data == "resultInterpreter":
			return LinkQualification.ResultInterpreter
		elif data == "request":
			return LinkQualification.Request
		else:
			raise Exception(f"{data} is not a valid value for LinkQualification enum.")

__all__ = ['LinkQualification']

File: cardinal-sdk/src/commonMain/kotlin/com/icure/cardinal/sdk/model/base/LinkQualification.kt
package com.icure.cardinal.sdk.model.base

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.String

@Serializable
public enum class LinkQualification(
	internal val dtoSerialName: String,
) {
	@SerialName("exact")
	Exact("exact"),

	@SerialName("narrower")
	Narrower("narrower"),

	@SerialName("broader")
	Broader("broader"),

	@SerialName("approximate")
	Approximate("approximate"),

	@SerialName("sequence")
	Sequence("sequence"),

	@SerialName("parent")
	Parent("parent"),

	@SerialName("child")
	Child("child"),

	@SerialName("relatedCode")
	RelatedCode("relatedCode"),

	@SerialName("linkedPackage")
	LinkedPackage("linkedPackage"),

	@SerialName("relatedService")
	RelatedService("relatedService"),

	@SerialName("inResponseTo")
	InResponseTo("inResponseTo"),

	@SerialName("replaces")
	Replaces("replaces"),

	@SerialName("transforms")
	Transforms("transforms"),

	@SerialName("transformsAndReplaces")
	TransformsAndReplaces("transformsAndReplaces"),

	@SerialName("appendsTo")
	AppendsTo("appendsTo"),

	@SerialName("basedOn")
	BasedOn("basedOn"),

	@SerialName("derivedFrom")
	DerivedFrom("derivedFrom"),

	@SerialName("device")
	Device("device"),

	@SerialName("focus")
	Focus("focus"),

	@SerialName("hasMember")
	HasMember("hasMember"),

	@SerialName("performer")
	Performer("performer"),

	@SerialName("specimen")
	Specimen("specimen"),

	@SerialName("resultInterpreter")
	ResultInterpreter("resultInterpreter"),

	@SerialName("request")
	Request("request"),
}
